docs/TypeScript.md

```diff
@@ -332,7 +332,7 @@ const todo = {
     done: { type: 'boolean' },
```

Comment:
Can you add somewhere an link to the `Type-Providers` document? I think that's valuable for the Typescript section.

Reply:
The Type Provider documentation probably belongs inside the `TypeScript.md` document directly, but I have included it as a separate document for future copy-writers to reference when updating documentation for Fastify version 4. 

There is a bit of information overlap with respect to utilizing TypeBox and json-schema-to-ts which may need to be updated inline with version 4 functionality, so would prefer to defer those edits on `TypeScript.md` for the time being.

Reply:
I'll create an issue for that when released then.